From 4beb72a142170ada4c8930835e133076eef15079 Mon Sep 17 00:00:00 2001
From: Dongwon Kim <dongwon.kim@intel.com>
Date: Thu, 26 May 2022 08:17:59 -0700
Subject: [PATCH 1/8] ui/gtk: detach VCS for additional guest displays
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Detaching any addtional guest displays in case there are multiple
displays assigned to the guest OS (e.g. max_outputs=n) so that
all of them are visible upon lauching.

Cc: Daniel P. Berrangé <berrange@redhat.com>
Cc: Markus Armbruster <armbru@redhat.com>
Cc: Philippe Mathieu-Daudé <philmd@redhat.com>
Cc: Paolo Bonzini <pbonzini@redhat.com>
Cc: Gerd Hoffmann <kraxel@redhat.com>
Cc: Vivek Kasireddy <vivek.kasireddy@intel.com>
Signed-off-by: Dongwon Kim <dongwon.kim@intel.com>
---
 ui/gtk.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/ui/gtk.c b/ui/gtk.c
index 1467b8c7d7..a3fc507e90 100644
--- a/ui/gtk.c
+++ b/ui/gtk.c
@@ -2311,6 +2311,8 @@ static void gtk_display_init(DisplayState *ds, DisplayOptions *opts)
     GtkDisplayState *s = g_malloc0(sizeof(*s));
     GdkDisplay *window_display;
     GtkIconTheme *theme;
+    int n_gfx_vcs = 0;
+    int i;
     char *dir;
 
     if (!gtkinit) {
@@ -2381,7 +2383,14 @@ static void gtk_display_init(DisplayState *ds, DisplayOptions *opts)
     gtk_widget_set_sensitive(s->copy_item,
                              vc && vc->type == GD_VC_VTE);
 #endif
-
+    for (i = 0; i < s->nb_vcs; i++) {
+        if (qemu_console_is_graphic(s->vc[i].gfx.dcl.con)) {
+            if (n_gfx_vcs > 0) {
+                gtk_menu_item_activate(GTK_MENU_ITEM(s->untabify_item));
+            }
+            n_gfx_vcs++;
+        }
+    }
     if (opts->has_full_screen &&
         opts->full_screen) {
         gtk_menu_item_activate(GTK_MENU_ITEM(s->full_screen_item));
-- 
2.20.1

